databricks/resource_databricks_dbfs_file.go

```diff
@@ -51,36 +70,56 @@ func resourceDBFSFile() *schema.Resource {
 func resourceDBFSFileCreate(d *schema.ResourceData, m interface{}) error {
 	client := m.(*service.DBApiClient)
 	path := d.Get("path").(string)
-	content := d.Get("content").(string)
 	overwrite := d.Get("overwrite").(bool)
+	//checksum := d.Get("content_b64_md5").(string)
 	mkdirs := d.Get("mkdirs").(bool)
 
 	if mkdirs {
-		parentDir := filepath.Dir(path)
-		err := client.DBFS().Mkdirs(parentDir)
+		parentDir, err := GetParentDirPath(path)
```

Comment:
(product code) this condition branch is about the side of the screen, so i'd suggest to move it to it's own dedicated function for redability